### PR TITLE
[OPS-8583] Drupal 9 update

### PR DESCRIPTION
**Pull request by orakili**
Refs: OPS-8583

This PR updates the starterkit repository with the latest standards for a Drupal 9 site based on the most recent sites.

1. Updated the list of dependencies with the modules commonly used in recent sites
2. Added list of recommended modules in README (ex: paragraphs)
3. Updated the tests (PHP unit and CS config + travis + script to run them locally)
4. Updated github workflows
5. Updated default config with sensible defaults and enabled modules
6. Updated pre-commit git hooks
7. Added script to replace `starterkit` etc. with the project name, site name and site URL
8. Added a minimal local stack, that serves 2 purposes: it can be used to create a site with the `starterkit` repo for example to edit the default site configuration and export it, and as a starting point until a more proper stack is set up.
9. Updated the README with more information to get started

**TODO**
- Add back `config_split`
- Review Drupal 10 compatibility and create new PR for that.

### CODE REVIEW

**Comment by orakili** [comment]
@lazysoundsystem Ready for your review. Hopefully the README contains all the necessary information but basically it just a matter of running `composer create-project unocha/starterkit` and then running `scripts/setup.sh` and optionally `local/install.sh`.

I'm still working on adding back `config_split` and also there will be another PR to use Drupal 10 and PHP 8.1.

**Review by lazysoundsystem — APPROVED** [review]
Looks great and will be really useful as a reference for the standard way we do things. I tried the local setup and it's very quick and easy. I'll adapt the setup script for the local stack setup too.)

Tagging in @attiks @left23 @rupl for information / feedback / testing.

**Comment by left23** [comment]
> Looks great and will be really useful as a reference for the standard way we do things. I tried the local setup and it's very quick and easy. I'll adapt the setup script for the local stack setup too.)
> 
> Tagging in @attiks @left23 @rupl for information / feedback / testing.

Is it https://github.com/UN-OCHA/local-reverse-proxy/tree/main/new-site-template I should be using for the stack? I can't find an equivalent `starterkit-stack` but I'm not sure if I don't have access or if it doesn't exist because `new-site-template` does this job?

File: README.md [hunk 1]
```diff
@@ -1,7 +1,134 @@
...
+
+Edit the following files, replacing `starterkit` with your project name (ex: `my-website`):
+
+- [.github/workflows/docker-build-image.yml](.github/workflows/docker-build-image.yml)
+
+### Docker
+
+Edit the following files:
+
+- [docker/Dockerfile](docker/Dockerfile) --> change `starterkit.test` to your **production** site URL.
```
[the remark this answers is not in the document]
Comment (orakili) [reply README.md:39]: Nice catch, thanks.

**Pull request merged**